### PR TITLE
[BugFix] fix group_concat for empty column

### PR DESCRIPTION
**Pull request by before-Sunrise**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

sql like "SELECT GROUP_CONCAT(lo_shipmode) orgs FROM lineorder WHERE 1 = 2;" will cause be crash
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: test/sql/test_agg_function/T/test_group_concat [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+PARTITION BY RANGE(`lo_orderdate`)
+(
+    PARTITION p1 VALUES [("-2147483648"), ("19930101")),
+    PARTITION p2 VALUES [("19930101"), ("19940101")),
+    PARTITION p3 VALUES [("19940101"), ("19950101")),
+    PARTITION p4 VALUES [("19950101"), ("19960101")),
+    PARTITION p5 VALUES [("19960101"), ("19970101")),
+    PARTITION p6 VALUES [("19970101"), ("19980101")),
+    PARTITION p7 VALUES [("19980101"), ("19990101")))
+DISTRIBUTED BY HASH(`lo_orderkey`) BUCKETS 48
```
Comment (imay) [inline test/sql/test_agg_function/T/test_group_concat:32]: Is it necessary to create so many tablets to cover this case?

File: test/sql/test_agg_function/T/test_group_concat [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+PROPERTIES (
+    "replication_num" = "1",
+    "colocate_with" = "groupa1",
+    "in_memory" = "false",
+    "storage_format" = "DEFAULT"
```
Comment (imay) [inline test/sql/test_agg_function/T/test_group_concat:37]: You can remove these two lines.

File: test/sql/test_agg_function/T/test_group_concat (hunk 1 above)
Comment (before-Sunrise) [reply test/sql/test_agg_function/T/test_group_concat:32]: > Is it necessary to create so many tablets to cover this case?
fixed

**Review by imay — APPROVED** [review]

File: test/sql/test_agg_function/R/test_group_concat [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+DISTRIBUTED BY HASH(`lo_orderkey`) BUCKETS 48
+PROPERTIES (
+    "replication_num" = "1"
+);
+-- result:
+-- !result
+SELECT GROUP_CONCAT(lo_shipmode) orgs FROM lineorder WHERE 1 = 2;
+-- result:
+None
+-- !result
```
Comment (ZiheLiu) [inline test/sql/test_agg_function/R/test_group_concat:17]: This SQL should return Empty not `None`?

```
-- result:
-- !result
```

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**